src/job_token/windows.rs

```diff
@@ -0,0 +1,67 @@
+use std::{
+    ffi::{CString, OsString},
+    io, ptr,
+};
+
+use crate::windows_sys::{
+    OpenSemaphoreA, ReleaseSemaphore, WaitForSingleObject, FALSE, HANDLE, SEMAPHORE_MODIFY_STATE,
+    THREAD_SYNCHRONIZE, WAIT_ABANDONED, WAIT_FAILED, WAIT_OBJECT_0, WAIT_TIMEOUT,
+};
+
+const WAIT_ABANDOEND_ERR_MSG: &str = r#" The specified object is a mutex object that was not released by the thread that owned the mutex object before the owning thread terminated. Ownership of the mutex object is granted to the calling thread and the mutex state is set to nonsignaled.
+
+If the mutex was protecting persistent state information, you should check it for consistency."#;
+
+pub(super) struct JobServerClient {
+    sem: HANDLE,
+}
+
+unsafe impl Sync for JobServerClient {}
+unsafe impl Send for JobServerClient {}
+
+impl JobServerClient {
+    pub(super) unsafe fn open(var: OsString) -> Option<Self> {
+        let s = var
+            .to_str()?
+            .split_ascii_whitespace()
+            .filter_map(|arg| {
+                arg.strip_prefix("--jobserver-fds=")
```

Comment:
On windows I think we should probably check `--jobserver-auth` first, and possibly not check `--jobserver-fds` at all. That's what https://www.gnu.org/software/make/manual/html_node/Windows-Jobserver.html implies, anyway. Maybe I'm missing some older docs that say that it can be provided in --jobserver-fds.

Reply:
jobserver implementation always checks both so I assumed that both can be used https://docs.rs/jobserver/latest/x86_64-pc-windows-msvc/src/jobserver/lib.rs.html#282

It actually makes sense that only `--jobserver-auth` is used since windows doesn't have to concept of fd anyway.

I will remove checking for `--jobswerver-fds` on windows

Reply:
Done.